### DISCUSSION TITLE
Is there any consideration for mqtt v5 compatible development?

### DISCUSSION THREAD

**Discussion by renjunok** [adam-fowler/mqtt-nio#30] [Ideas]
V5 can customize the error type returned, and the requirements are quite common.

**Comment by adam-fowler** [comment]
Currently I am using an MQTT lib from amazon for the packet parsing and that is limited to v3.1.1. I have considered moving to my own parsing and at that point it would be a lot easier for me to consider other versions but at this point in time it wouldnt be that easy

**Reply by renjunok** [reply]
Understand, thank you for your library to make my App run normally.

😊

**Reply by adam-fowler** [reply]
@renjunok I moved the MQTT packet serialization to swift recently and thought I could take on a v5 client. Would you be interested in testing it out. You can find the PR here #39.

Create a client as follows 
```swift
let client = MQTTClient(
    host: host,
    identifier: "MyAWSClient",
    eventLoopGroupProvider: .createNew,
    configuration: .init(version: .v5_0)
)
```

You can use the standard calls as you would before but now there are V5 specific calls which allow for property setting and reading and returning the additional return codes. The v5 calls are accessed via `MQTTClient.v5` 
```swift
let connack = client.v5.connect()
```

**Comment by adam-fowler** [comment]
v2.0.0 of MQTTNIO has been released with v5.0 support

**Reply by renjunok** [reply]
very good!😊
